api/schema/openapi.yaml

```diff
@@ -0,0 +1,305 @@
+          description: Internal server error
+
+  /api/v1/owner/redirect:
+    get:
+      summary: Get Owner Information
+      operationId: getOwnerInfo
+      responses:
+        '200':
+          description: Successful response with Owner Information
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OwnerInfo'
+        '404':
+          description: No Owner Information found
+        '500':
+          description: Internal server error
+    post:
+      summary: Create Owner Information
+      operationId: createOwnerInfo
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/OwnerInfo'
+          text/plain:
+            schema:
+              type: string
+              format: byte # Assuming text/plain is used for raw bytes which decode to JSON
+      responses:
+        '201':
+          description: Owner Information created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OwnerInfo'
+        '400':
+          description: Invalid input
+        '409':
+          description: Owner Information already exists
+        '500':
+          description: Internal server error
+    put:
+      summary: Update Owner Information
+      operationId: updateOwnerInfo
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/OwnerInfo'
+          text/plain:
+            schema:
+              type: string
+              format: byte # Assuming text/plain is used for raw bytes which decode to JSON
+      responses:
+        '200':
+          description: Owner Information updated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OwnerInfo'
+        '400':
+          description: Invalid input
+        '404':
+          description: No Owner Information found
+        '500':
+          description: Internal server error
+
+  /api/v1/vouchers:
+    get:
+      summary: Get Voucher by GUID
+      operationId: getVoucherByGuid
+      parameters:
+        - name: guid
+          in: query
+          required: true
+          description: The GUID of the voucher to retrieve (hexadecimal string).
+          schema:
+            $ref: '#/components/schemas/GUID'
+      responses:
+        '200':
+          description: Successful response with Voucher and Owner Keys
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  voucher:
+                    $ref: '#/components/schemas/Voucher'
+                  owner_keys:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/OwnerKey'
```

Comment:
There should be a "extend GUID to this owner pubkey" endpoint and a "get voucher (no private key in response" endpoint. These endpoints may have different authorizations.

Maintaining Java compatibility is not desirable here. I don't want to be sending private keys.

Reply:
@ben-krieger Agreed. These points need to be addressed and the whole API could use a good going over.

From a workflow perspective I'd propose that I open new issues for the above changes and address them in follow on patches.  My goal for this patch was twofold: create the initial openapi schema based on the current API - warts and all - and provide a set of initial verification test cases.  Landing that would give us a starting point for API refactor/improvement with the ability to validate the refactoring.

With that in mind I'd like to land this patch first, then follow up with API refactoring.  Does that make sense?

Reply:
Yep, please open a new issue referencing my comment and feel free to merge this PR in the meantime.

Reply:
Added issues #41 and #42 to track this work.